src/runtime/app.ts

```diff
@@ -36,6 +36,26 @@ function createNitroApp (): NitroApp {
 
   const routerOptions = createMatcher({ routes: config.nitro.routes })
 
+  h3App.use(eventHandler((event) => {
+    const routeOptions = routerOptions.lookup(event.req.url) || {}
```

Comment:
Note: lookup only looks up to the first matching tree node which is an issue (will reproduce later). We need to fix it asap before marking routing rules as stable...

Reply:
I am aware, and mentioned it in rc12 release notes.

Reply:
#582